core/trino-main/src/main/java/io/trino/operator/output/PagePartitioner.java

```diff
@@ -444,11 +446,16 @@ private Page getPartitionFunctionArguments(Page page)
         return new Page(page.getPositionCount(), blocks);
     }
 
+    @Override
     public void close()
     {
-        flushPositionsAppenders(true);
-        flushPageBuilders(true);
-        memoryContext.close();
+        try {
+            flushPositionsAppenders(true);
```

Comment:
Why would flush fail?

Reply:
e.g. there may be a bug for a specific query like `PossitionsAppender` assuming incorrectly the block type like in https://github.com/trinodb/trino/pull/15948